packages/ketcher-core/src/application/editor/tools/Erase.ts

```diff
@@ -46,10 +48,28 @@ class EraserTool implements BaseTool {
     }
 
     if (selectedItemRenderer instanceof BaseRenderer) {
+      const clickedEntity = selectedItemRenderer.drawingEntity;
+
+      if (
+        clickedEntity instanceof BaseMonomer &&
+        this.editor.drawingEntitiesManager.selectedEntities.length
+      ) {
+        const unselectCommand = new Command();
+        clickedEntity.forEachBond((bond) => {
+          if (bond.selected) {
+            unselectCommand.merge(
+              this.editor.drawingEntitiesManager.unselectDrawingEntity(bond),
+            );
+          }
+        });
+        if (unselectCommand.operations.length) {
+          this.history.update(unselectCommand);
+          this.editor.renderersContainer.update(unselectCommand);
+        }
+      }
+
       const modelChanges =
-        this.editor.drawingEntitiesManager.deleteDrawingEntity(
-          selectedItemRenderer.drawingEntity,
-        );
+        this.editor.drawingEntitiesManager.deleteDrawingEntity(clickedEntity);
```

Comment:
Can we keep it as backup solution and think about the way of fixing that without unselecting it explicitly?